test/clj/game/core/abilities_test.clj

```diff
@@ -62,6 +62,20 @@
     (is (core/has-trash-ability? card)
         (str (:title card) " needs either :cost [(->c :trash-can)] or :trash-icon true"))))
 
+(deftest actions-are-documented
+  "This ensures that every ability with a (->c :click ...) cost is marked as either :action true, or :action (otherwise)
+   this is important for the functionality of undo-click, so it's worth ensuring"
+  (doseq [card (vals @all-cards)
+          :when (not-empty (card-def card))]
+    ;;(println ":card: " (:title card))
+    (doseq [{:keys [cost] :as ab} (apply concat (vals (select-keys (card-def card) [:abilities :corp-abilities :runner-abilities])))]
+      ;;(println ":card: has abilities? " (:title card))
+      (when (and cost
+               (or (and (sequential? cost)
+                        (some #(= (:cost/type %) :click) cost))
+                   (= (:cost/type cost) :click)))
+        (is (contains? ab :action) (str (:title card) " may have unlabelled actions (use :action true) or (:action false/nil)"))))))
```

Comment:
this can be one `doseq`: `[card ... :when ... {:keys [cost]} ... :when ...] (is (contains? ab ;action) ...)`